lib/improver/spotdata/extract_data.py

```diff
@@ -157,52 +159,111 @@ def process(self, cube, sites, neighbours, ancillary_data, additional_data,
             self.method, self.__class__.__name__))
 
     @staticmethod
-    def _build_coordinates(latitudes, longitudes, altitudes, utc_offsets,
-                           wmo_sites):
+    def _build_coordinate(data, coordinate, coord_type=DimCoord,
+                          data_type=float, units='1', bounds=None,
+                          coord_system=None, custom_function=None):
```

Comment:
There's a ticket for that, once it's agreed what is considered sufficiently general: #180